x-pack/plugins/security_solution/public/attack_discovery/hooks/use_poll_api.tsx

```diff
@@ -0,0 +1,130 @@
+  toasts?: IToasts;
+  connectorId?: string;
+}
+
+export interface AttackDiscoveryData extends AttackDiscoveryResponse {
+  connectorId: string;
+}
+
+export const usePollApi = ({
+  http,
+  setApproximateFutureTime,
+  toasts,
+  connectorId,
+}: Props): {
+  status: AttackDiscoveryStatus | null;
+  data: AttackDiscoveryData | null;
+  pollApi: () => void;
+} => {
+  const [status, setStatus] = useState<AttackDiscoveryStatus | null>(null);
+  const [data, setData] = useState<AttackDiscoveryData | null>(null);
+  const timeoutIdRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  useEffect(() => {
+    return () => {
+      // when a connectorId changes, clear timeout
+      if (timeoutIdRef.current) clearTimeout(timeoutIdRef.current);
+    };
+  }, [connectorId]);
+
+  const handleResponse = useCallback(
+    (responseData: AttackDiscoveryResponse) => {
+      if (connectorId == null || connectorId === '') {
+        throw new Error('Invalid connector id');
+      }
+      setStatus(responseData.status);
+      setApproximateFutureTime(
+        moment(responseData.updatedAt).add(responseData.averageIntervalMs, 'milliseconds').toDate()
+      );
+      setData({
+        ...responseData,
+        connectorId,
+        attackDiscoveries: responseData.attackDiscoveries.map((attackDiscovery) => ({
+          ...attackDiscovery,
+          id: attackDiscovery.id ?? uuid.v4(),
+          detailsMarkdown: replaceNewlineLiterals(attackDiscovery.detailsMarkdown),
+          entitySummaryMarkdown: replaceNewlineLiterals(attackDiscovery.entitySummaryMarkdown),
+          summaryMarkdown: replaceNewlineLiterals(attackDiscovery.summaryMarkdown),
+        })),
+      });
+    },
+    [connectorId, setApproximateFutureTime]
+  );
+  const pollApi = useCallback(async () => {
+    try {
+      if (connectorId == null || connectorId === '') {
+        throw new Error('Invalid connector id');
+      }
+      // call the internal API to generate attack discoveries:
+      const rawResponse = await http.fetch(
+        `/internal/elastic_assistant/attack_discovery/${connectorId}`,
+        {
+          method: 'GET',
+          version: ELASTIC_AI_ASSISTANT_INTERNAL_API_VERSION,
+        }
+      );
+
+      const parsedResponse = AttackDiscoveryGetResponse.safeParse(rawResponse);
+      if (!parsedResponse.success) {
+        throw new Error('Failed to parse the attack discovery GET response');
+      }
+      if (parsedResponse.data.entryExists === false || parsedResponse.data.data == null) {
+        setStatus(null);
+        setData(null);
+      } else if (
+        parsedResponse.data.data.status === attackDiscoveryStatus.succeeded ||
+        parsedResponse.data.data.status === attackDiscoveryStatus.failed
+      ) {
+        handleResponse(parsedResponse.data.data);
+      } else if (parsedResponse.data.data.status === attackDiscoveryStatus.running) {
+        handleResponse(parsedResponse.data.data);
```

Comment:
Consider refactoring the conditional code below:

```typescript
      } else if (
        parsedResponse.data.data.status === attackDiscoveryStatus.succeeded ||
        parsedResponse.data.data.status === attackDiscoveryStatus.failed
      ) {
        handleResponse(parsedResponse.data.data);
      } else if (parsedResponse.data.data.status === attackDiscoveryStatus.running) {
        handleResponse(parsedResponse.data.data);
```

if conditional execution is not required, because `handleResponse(parsedResponse.data.data);` appears to be called regardless of the status (`succeeded`, `failed`, or `running`).